packages/plugin-relay/src/utils/internal.ts

```diff
@@ -18,10 +19,28 @@ export function internalDecodeGlobalID<Types extends SchemaTypes>(
   builder: PothosSchemaTypes.SchemaBuilder<Types>,
   globalID: string,
   ctx: object,
+  info: GraphQLResolveInfo,
+  types?: string[] | null,
 ) {
-  if (builder.options.relayOptions.decodeGlobalID) {
-    return builder.options.relayOptions.decodeGlobalID(globalID, ctx);
+  const decoded = builder.options.relayOptions.decodeGlobalID
+    ? builder.options.relayOptions.decodeGlobalID(globalID, ctx)
+    : decodeGlobalID(globalID);
+
+  if (types && !types.includes(decoded.typename)) {
+    throw new Error(`ID: ${globalID} is not of type: ${types.join(', ')}`);
```

Comment:
A bit off topic from the rest of the work here (which looks great), but something I was thinking about regarding how this might be used as an API validation layer.

Wondering whether it'd be helpful to throw framework logic errors like this with a `PothosError` class which extends `Error`, so it'd be possible to distinguish it from other types of internal application errors that might happen in a resolver which you might not want to expose publicly.

Reply:
Yes! this has been on my todo list for a while.  I really want to create a plugable/configurable error system in Pothos so customizing error messages is easy, but having all pothos error sub-class a core PothosError class would be a good and easy to implement start.